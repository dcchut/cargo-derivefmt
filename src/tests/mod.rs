use textwrap::dedent;

use crate::modify_source;

#[test]
fn test_derive_ordering() {
    let mut source = dedent(
        r#"
        #[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
        struct Wrapped<T>(T);
    "#,
    );
    modify_source(&mut source).unwrap();
    insta::assert_snapshot!(source);
}

#[test]
fn test_derive_ordering_qualified() {
    let mut source = dedent(
        r#"
        #[derive(Clone, Copy, std :: fmt :: Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
        struct SillyBugger;
    "#,
    );
    modify_source(&mut source).unwrap();
    insta::assert_snapshot!(source);
}
