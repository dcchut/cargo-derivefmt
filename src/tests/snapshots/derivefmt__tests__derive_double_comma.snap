---
source: src/tests/mod.rs
expression: source
---

#[derive(Apples, Clone, Debug, Default, Hash)]
struct T;

